### DISCUSSION TITLE
do you need a maincraft license to connect to the cloud?

### DISCUSSION THREAD

**Discussion by DOOMSENKO** [Gakuto1112/SenkoSan#19] [Q&A] [answered]
I'm playing with tlauncher and noticed that there is no connection to the cloud and I can't change the costume for senko or other characters because the avatar catalog writes (not yet available)

**Comment by Gakuto1112** [comment, marked as answer]
Yes, you need to buy an official Minecraft license to use the Figura backend system (Uploading your avatar, Seeing other players' avatars).

The Figura backend system has a authentication system that requires an official Minecraft license. And, as you know, the Minecraft game launched by TLauncher doesn't have the license. So, it is impossible to use the Figura backend system without having the license.

This also applies for the case that you join a server where the online mode is disabled.

From [Figura README](https://github.com/FiguraMC/Figura?tab=readme-ov-file#-my-minecraft-is-cracked-non-premiumnon-original-or-im-trying-to-join-a-cracked-offline-mode-server-why-cant-i-use-figura)
> - ### My Minecraft is cracked (non-premium/non-original) or I'm trying to join a cracked offline mode server, why can't I use Figura?
> > Figura uses your account's UUID and your Mojang authentication as a way to prove you own that account, avoiding unwanted / malicious uploads
> > 
> > Non-premium Minecraft accounts don't authenticate with Mojang, and Offline mode servers don't report working UUID's, as such can neither upload nor download Figura avatars
